src/workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,134 @@
+import {P5Visualizer} from '../visualizers/P5Visualizer'
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+
+/** md
+# p5 Visualizer Template
+
+[image should go here]
+
+This template can be used as a starting point for designing a p5 visualizer.
+
+## Parameters
+**/
+
+class P5VisualizerTemplate extends P5Visualizer {
+    name = 'p5 Visualizer Template' // the name shown on the visualizer menu
+
+    // external parameters, which the user can choose while creating the
+    // visualizer bundle
+    baseIndex = 0 // the default value of this parameter is zero
+    params = {
+        /** md
+- baseIndex: Where in the sequence to start browsing.
+         **/
+        baseIndex: {
+            value: this.baseIndex,
+            displayName: 'Base index',
+            required: true,
+        },
+    }
+
+    // internal parameters. the value of `index` will always be overwritten in
+    // `setup`, but TypeScript can't infer that, so we have to give `index` an
+    // initial value
+    index = 0
+    flash = 0
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        // make sure the base index doesn't precede the first index
+        if (this.params.baseIndex.value < this.seq.first) {
+            status.isValid = false
+            status.errors.push(
+                "The base index can't precede the"
+                    + ` first index, ${this.seq.first}`
+            )
+        }
+
+        return status
+    }
+
+    setup() {
+        // run the setup process for a general p5 visualizer, and make sure the
+        // p5 sketch has been successfully created. most p5 visualizers should
+        // do this
+        super.setup()
+        if (!this.sketch) {
+            throw (
+                'Attempt to show p5 Template Visualizer'
+                + ' before injecting into element'
+            )
+        }
+
+        // start viewing the sequence at the base index, which the user chose
+        // when creating the visualizer
+        this.index = this.baseIndex
+
+        this.sketch.stroke(51, 51, 255)
+        this.sketch.textAlign(this.sketch.CENTER, this.sketch.CENTER)
+        /*this.sketch.fill(128, 159, 255)*/
+    }
+
+    draw() {
+        super.draw()
+
+        // for convenience, give the sketch a shorter name
+        const sketch = this.sketch
```

Comment:
Also saves some (miniscule amount) of time in avoiding repeated error checking generated by access to this.sketch. Not sure whether or not you want to mention this.

Reply:
I've revised the comments and visualizer-building guide to emphasize this.